src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -83,9 +85,39 @@ std::shared_ptr<ov::Model> add_slices_to_kvcache_inputs(const std::shared_ptr<ov
     return std::make_shared<ov::Model>(model->get_results(), ov::SinkVector{}, new_params);
 }
 
+struct KVAxesPosition {
+    uint32_t batch;
+    uint32_t seq_len;
+};
+
+KVAxesPosition get_kv_axes(const std::string& model_type) {
+    KVAxesPosition axes;
+    if (model_type == "chatglm") {
+        axes.batch = 1u;
+        axes.seq_len = 0u;
```

Comment:
why this approach https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/cpp/prompt_lookup_decoding_lm/prompt_lookup_decoding_lm.cpp#L13 is not used? it seems generic and does not required model type

Reply:
@ilya-lavrenov Yes, you're right. The only drawback is that this function cannot be used for `.blob`'s case as they will be stateless + static.

Reply:
how is it supposed to pass compiled blob via current LLMPipeline API ?

Reply:
It's on review so far: https://github.com/openvinotoolkit/openvino.genai/pull/811

Reply:
The current solution aim to handle `qwen` / `chatglm` case that are crucial for now.

But in general, I'd prefer using your approach but somewhere inside `StatefulToStateless` transformation, so that it could save necessary `metadata` that will be available from both `xml` / `blob` formats.